src/toil/batchSystems/kubernetes.py

```diff
@@ -49,13 +49,21 @@
                                                    BatchSystemLocalSupport)
 from toil.lib.humanize import human2bytes
 from toil.resource import Resource
-from toil.lib.retry import retry_http
+
+from toil.lib.retry import retry
 
 logger = logging.getLogger(__name__)
 
+def exception_found(e):
+    return e.error_code.endswith('.ApiException') or e.error_code.endswith('.MaxRetryError')
```

Comment:
You might still want to enforce that only some API exceptions (GATEWAY_TIMEOUT, MAX_RETRY_ERROR) are recognized here. We don't want to retry when our code makes a genuine mistake using the API. If we try to make a job in a nonexistent namespace, for example, we should just fail right away, I think, instead of retrying a bunch.